class.jetpack-settings-sync.php

```diff
@@ -0,0 +1,338 @@
+<?php
+
+/**
+ * All the Settings we want to sync to .com
+ *
+ * This is also used when displaying settings in the API
+ */
+class Jetpack_Sync_Settings {
+
+	private static $options = array(
+
+		'blogname'                     	 => 'string',
+		'blogdescription'              	 => 'string',
+
+		'default_ping_status'          	 => 'closed',
+		'default_comment_status'       	 => 'closed',
+		'blog_public'                  	 => 'string',
+
+		'infinite_scroll'              	 => 'bool',
+		'default_category'             	 => 'int',
+		'default_post_format'          	 => 'string',
+		'default_pingback_flag'          => 'bool',
+		'require_name_email'           	 => 'bool',
+		'comment_registration'         	 => 'bool',
+		'close_comments_for_old_posts' 	 => 'bool',
+		'close_comments_days_old'      	 => 'int',
+		'thread_comments'              	 => 'bool',
+		'thread_comments_depth'        	 => 'int',
+		'page_comments'                	 => 'bool',
+		'comments_per_page'            	 => 'int',
+		'default_comments_page'        	 => array( 'newest', 'oldest' ),
+		'comment_order'                	 => array( 'asc', 'desc' ),
+		'comments_notify'              	 => 'bool',
+		'moderation_notify'            	 => 'bool',
+		'social_notifications_like'    	 => 'on',
+		'social_notifications_reblog'  	 => 'on',
+		'social_notifications_subscribe' => 'on',
+		'comment_moderation'           	 => 'bool',
+		'comment_whitelist'            	 => 'bool',
+		'comment_max_links'            	 => 'int',
+		'moderation_keys'              	 => 'string',
+		'blacklist_keys'               	 => 'string',
+		'lang_id'                      	 => 'int',
+		'wga'                          	 => array( 'code'=>'regex:/^$|^UA-[\d-]+$/i' ),
+		'disabled_likes'               	 => 'bool',
+		'disabled_reblogs'             	 => 'bool',
+		'jetpack_comment_likes_enabled'	 => 'bool',
+		'twitter_via'                  	 => 'string',
+		'jetpack-twitter-cards-site-tag' => 'string',
+		/*
+		'sharing_button_style'         	 => 'string',
+		'sharing_label'                	 => 'string',
+		'sharing_show'                 	 => 'string',
+		'sharing_open_links'           	 => 'string',
+		*/
+	);
+
+	/**
+	 * Data that we want to sync.
+	 * @var array
+	 */
+	private static $mock_options = array(
+		'admin_url' => array(
+			'type' => 'url',
+			'callback' => 'get_admin_url'
+		),
+		'jetpack_protect_whitelist' => array(
+			'type' => array( 'local' => 'array', 'global' => 'array' ),
+			'callback' => 'jetpack_protect_format_whitelist'
+		),
+		'post_categories' => array(
+			'type' => 'array',
+			'callback' => array( 'Jetpack_Sync_Settings', 'get_post_caregories' )
+		),
+		'jetpack_sync_non_public_post_stati' => array(
+			'type' => 'bool',
+			'callback' => array( 'Jetpack_Options','get_option' ),
+			'callback_args' => array( 'sync_non_public_post_stati' )
+		),
+
```

Comment:
two empty lines, it would be good to homogenise the number of empty lines between `$mock_options`